src/bvals/cc/bvals_cc_in_one.cpp

```diff
@@ -0,0 +1,553 @@
+  sj = (nb.ni.ox2 > 0) ? (cellbounds.je(interior) - cn) : cellbounds.js(interior);
+  ej = (nb.ni.ox2 < 0) ? (cellbounds.js(interior) + cn) : cellbounds.je(interior);
+  sk = (nb.ni.ox3 > 0) ? (cellbounds.ke(interior) - cn) : cellbounds.ks(interior);
+  ek = (nb.ni.ox3 < 0) ? (cellbounds.ks(interior) + cn) : cellbounds.ke(interior);
+
+  // send the data first and later prolongate on the target block
+  // need to add edges for faces, add corners for edges
+  if (nb.ni.ox1 == 0) {
+    if (nb.ni.fi1 == 1)
+      si += pmb->block_size.nx1 / 2 - pmb->cnghost;
+    else
+      ei -= pmb->block_size.nx1 / 2 - pmb->cnghost;
+  }
+  if (nb.ni.ox2 == 0 && pmb->block_size.nx2 > 1) {
+    if (nb.ni.ox1 != 0) {
+      if (nb.ni.fi1 == 1)
+        sj += pmb->block_size.nx2 / 2 - pmb->cnghost;
+      else
+        ej -= pmb->block_size.nx2 / 2 - pmb->cnghost;
+    } else {
+      if (nb.ni.fi2 == 1)
+        sj += pmb->block_size.nx2 / 2 - pmb->cnghost;
+      else
+        ej -= pmb->block_size.nx2 / 2 - pmb->cnghost;
+    }
+  }
+  if (nb.ni.ox3 == 0 && pmb->block_size.nx3 > 1) {
+    if (nb.ni.ox1 != 0 && nb.ni.ox2 != 0) {
+      if (nb.ni.fi1 == 1)
+        sk += pmb->block_size.nx3 / 2 - pmb->cnghost;
+      else
+        ek -= pmb->block_size.nx3 / 2 - pmb->cnghost;
+    } else {
+      if (nb.ni.fi2 == 1)
+        sk += pmb->block_size.nx3 / 2 - pmb->cnghost;
+      else
+        ek -= pmb->block_size.nx3 / 2 - pmb->cnghost;
+    }
+  }
+}
+
+// send boundary buffers with MeshBlockPack support
+// TODO(pgrete) should probaly be moved to the bvals or interface folders
+TaskStatus SendBoundaryBuffers(std::shared_ptr<MeshData<Real>> &md) {
+  Kokkos::Profiling::pushRegion("Task_SendBoundaryBuffers_MeshData");
+
+  auto var_pack = md->PackVariables(std::vector<MetadataFlag>({Metadata::FillGhost}));
+  const auto Nv = var_pack.GetDim(4);
+
+  Kokkos::Profiling::pushRegion("Reset boundaries");
+
+  int buffers_used = 0;
+  // reset buffer and count required buffers
+  for (int block = 0; block < md->NumBlocks(); block++) {
+    auto &rc = md->GetBlockData(block);
+    auto pmb = rc->GetBlockPointer();
+
+    for (auto &v : rc->GetCellVariableVector()) {
+      if (v->IsSet(parthenon::Metadata::FillGhost)) {
+        v->resetBoundary();
+      }
+    }
+
+    for (int n = 0; n < pmb->pbval->nneighbor; n++) {
+      parthenon::NeighborBlock &nb = pmb->pbval->neighbor[n];
+      // TODO(?) currently this only works for a single "Variable" per container.
+      // Need to update the buffer sizes so that it matches the packed Variables.
+      assert(rc->GetCellVariableVector().size() == 1);
+      auto *bd_var_ = rc->GetCellVariableVector()[0]->vbvar->GetPBdVar();
+      if (bd_var_->sflag[nb.bufid] == parthenon::BoundaryStatus::completed) continue;
+      buffers_used += 1;
+    }
+  }
+  Kokkos::Profiling::popRegion(); // Reset boundaries
+
+  auto boundary_info = md->GetSendBuffers();
+
+  if (!boundary_info.is_allocated() || true) {
```

Comment:
Is the intent this always true? That seems odd.

Reply:
That's the "bug" I was referring to and where I'd like additional eyes on.
With the `true` statement that caching of the `boundary_info` is effectively disabled (because the `boundary_info` array is recalculated for every function call).
Remove the `true` make use of the caching, but results in errors in the regression tests that I was not able to track down yet.
I use the same caching for the `boundary_info` in the set routine and there it works as expected.

Reply:
got it. I'll poke at it.